q2_fmt/_peds.py

```diff
@@ -466,11 +479,458 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
+    peds_iters: pd.Series
+        a pd.Series with all iterations of mismatched PEDS Values. This will
+        later be compared to an actual PEDS value.
+
+    Examples
+    --------
+    mismatched_peds = [0, 0, 0,0 ]
+    iterations = 10
+
+    _simulate_uniform_distro(mismatched_peds, iterations)
+
+    pd.Series([data = [0, 0, 0, 0, 0, 0, 0, 0, 0],
+               index = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]])
+    """
+    peds_iters = random.choices(mismatched_peds, k=iterations)
+    # Tranforming to series for easy series math in _per_subject_stats()
+    peds_iters = pd.Series(peds_iters)
+    return peds_iters
+
+
+def _per_subject_stats(mismatched_peds, actual_temp,
+                       iterations, mismatched_pairs_n):
+    """Creates per subject PEDS stats
+
+    Creates per subject PEDS stats by sampling mismatch PEDS values 'iteration'
+    number of time. The list of mismatched PEDS values is compared to the
+    actual PEDS value. The Test Statistic for this is the iterations - the
+    number of mismatched PEDS value that are greater than the actual p-value.
+    p-values are corrected by adding 1 to each side of the faction
+    because Monte Carlo Simulation is able to produce an invalid p-value of 0.
+
+    Parameters
+    ----------
+    mismatched_peds: pd.DataFrame
+        A Dataframe that contains all mismatched PEDS values.
+    actual_temp: pd.Series
+        A Series containing Sample IDs as the index and actual PEDS values for
+        the sample as the value.
+    iterations: int
+        Number of iterations to run simulations (Number of times to randomly
+        sample mismatched_peds)
+    mismatched_peds_n: int
+        Number of mismatched donor pairs are possible.
+
+    Returns
+    -------
+    per_subject_stats: pd.DataFrame
+        a Dataframe containing test statistics, p-values, q-values and
+        sample size info. Note: this is the DataFrame that is returned
+        to the user as the per-subject-stats.qza
+
+    Examples
+    --------
+    mismatched_peds = [0, 0, 0, 0]
+    actual_temp = pd.Series([data = [1, 1, 1, 1],
+               index = [sample1, sample2, sample3, sample4]])
+    iterations = 10
+    mismatched_pairs_n = 4
+
+    _per_subject_stats(mismatched_peds, actual_temp,
+                       iterations, mismatched_pairs_n):
+
+    pd.Dataframe({"A:group": [sample1, sample2, sample3, sample4],
+                   "A:n": [1, 1, 1, 1],
+                   "A:measure" : [1, 1, 1, 1],
+                   "B:group" : [shuffled_recipients, shuffled_recipients,
+                                shuffled_recipients, shuffled_recipients],
+                   "B:n": [4, 4, 4, 4],
+                   "B:measure": [0, 0, 0, 0],
+                   "n": [10, 10, 10, 10],
+                   "test-statistic": [10, 10, 10, 10],
+                   "p-value": [0.001, 0.001, 0.001, 0.001],
+                   "q-value": [0.004, 0.002, 0.00133, 0.001]})
+    """
+    disagree_df = pd.DataFrame([])
```

Comment:
If you like the `disagree` nomenclature here, I would add a comment describing it so it's clearer to readers of the code. I typically use something like `count_greater` or `count_gte` (for greater than or equal), but your call.